cmip6_downscaling/analysis/analysis.py

```diff
@@ -0,0 +1,183 @@
+import os
+
+import cartopy.crs as ccrs
+import fsspec
+import matplotlib.pyplot as plt
+import pandas as pd
+import papermill as pm
+import xarray as xr
+from azure.storage.blob import BlobServiceClient, ContentSettings
+from prefect import task
+
+from cmip6_downscaling.data.cmip import convert_to_360
+
+from .qaqc import make_qaqc_ds
+
+connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+fs = fsspec.filesystem('az', connection_string=connection_string)
+
+
+def qaqc_checks(ds):
+    # run_qaqc_dir =  f"{analysis_dir}/qaqc/{run_id}"
+    qaqc_ds = make_qaqc_ds(ds)
+    annual_qaqc_ts = qaqc_ds.groupby('time.year').sum().sum(dim=['lat', 'lon']).to_dataframe()
+    qaqc_maps = qaqc_ds.sum(dim='time')
+    return annual_qaqc_ts, qaqc_maps
+
+
+def monthly_summary(ds):
+    """Creates an monthly summary dataset.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Downscaled dataset
+    """
+    out_ds = xr.Dataset()
+    for var in ds:
+        if var in ['tasmax', 'tasmin']:
+            out_ds[var] = ds[var].resample(time='1MS').mean(dim='time')
+        elif var in ['pr']:
+            out_ds[var] = ds[var].resample(time='1MS').sum(dim='time')
+        else:
+            print(f'{var} not implemented')
+
+    return out_ds
+
+
+def annual_summary(ds):
```

Comment:
```suggestion
def annual_summary(ds: xr.Dataset) -> xr.Dataset:
```

also add return section to docstring